assets/main.cpp

```diff
@@ -16,30 +16,182 @@ si::MemoryBuffer mxHd;
 
 void CreateWidescreen()
 {
+	std::string result = out + "/widescreen.si";
+	struct AssetView {
+		std::string name;
+		std::string extra;
+	};
+	const AssetView widescreenBitmaps[] = {
+		{"GaraDoor_Background_Wide",
+		 "World:current, StartWith:\\Lego\\Scripts\\Isle\\Isle;1160, RemoveWith:\\Lego\\Scripts\\Isle\\Isle;1161"}
+	};
+
 	si::Interleaf si;
-	std::string file = out + "/widescreen.si";
 	mxHd.seek(0, si::MemoryBuffer::SeekStart);
 	si.Read(&mxHd);
 
-	si::Object GaraDoor_Wide;
-	const char extra[] =
-		"World:current, StartWith:\\Lego\\Scripts\\Isle\\Isle;1160, RemoveWith:\\Lego\\Scripts\\Isle\\Isle;1161";
-	GaraDoor_Wide.type_ = si::MxOb::Bitmap;
-	GaraDoor_Wide.flags_ = MxDSAction::c_enabled | MxDSAction::c_bit4;
-	GaraDoor_Wide.duration_ = -1;
-	GaraDoor_Wide.loops_ = 1;
-	GaraDoor_Wide.extra_ = si::bytearray(extra, sizeof(extra));
-	GaraDoor_Wide.presenter_ = "MxStillPresenter";
-	GaraDoor_Wide.name_ = "GaraDoor_Wide";
-	GaraDoor_Wide.filetype_ = si::MxOb::STL;
-	GaraDoor_Wide.location_.x = -240.0;
-	GaraDoor_Wide.location_.z = -1.0;
-	GaraDoor_Wide.up_.y = 1.0;
-	GaraDoor_Wide.ReplaceWithFile("widescreen/garadoor.bmp");
-	si.AppendChild(&GaraDoor_Wide);
-	depfile << file << ": " << (std::filesystem::current_path() / "widescreen/garadoor.bmp").string() << std::endl;
-
-	si.Write(file.c_str());
+	int i = 0;
+	for (const AssetView& asset : widescreenBitmaps) {
+		si::Object* object = new si::Object;
+		std::string file = std::string("widescreen/") + asset.name + ".bmp";
```

Comment:
Does and should this program verify whether the bmp's and wav's are actually there, and are no LFS text files?

Reply:
This should take care of it:

```
		if (!object->ReplaceWithFile(file.c_str())) {
			abort();
		}
```

If it's not a valid file according to the specified type this should always abort

Reply:
My only remark is to print something to stderr as feedback.

I made this comment first thinking an ordinary `git fetch` or `git pull` would not download lfs files, but apparently they are cloned automatically.